src/main/java/services/petshop/steps/CommonMethodsPetSteps.java

```diff
@@ -7,34 +7,41 @@
 import net.thucydides.core.annotations.Step;
 import net.thucydides.core.annotations.Steps;
 import services.petshop.pojo.requests.Pet;
+import utils.methods.ReusableMethods;
 
 public class CommonMethodsPetSteps {
 
+    ReusableMethods reusableMethods = new ReusableMethods();
+
     @Steps
     PostPetSteps postPetSteps;
 
     @Steps
     GetPetSteps getPetSteps;
 
-    @Step("When I retrieve the id of the created pet")
-    public void getTheIdOfTheCreatedPet(Pet mypet){
+    @Step("When I create a new pet and retrieve the id of the created pet")
+    public int createPetAndGetItsId(Pet mypet) {
         Response createPetR = postPetSteps.createNewPet(mypet);
         Pet createdPet = createPetR.as(Pet.class);
-        System.out.println(getPetSteps.getPetWithId(createdPet.getId()));
+        return createdPet.getId();
     }
 
-    @Step("Then I check the status code of the response when I perform an action successfully is 200")
-    public void statusCodeSuccess(Response response){
-        Assert.assertEquals("The status is not 200 OK.",200, response.getStatusCode());
+    @Step("Then I check the status code of the response when I perform this action is {1}")
+    public void statusCode(Response response, int statusCode) {
+        reusableMethods.validateResponseStatusCode(response, statusCode);
     }
 
-    @Step("Assert the user that has been retrieved with the get method is the same user that has been created")
-    public void assertUserRetrievedIsUserCreated(Pet mypet){
-        Response createPetR = postPetSteps.createNewPet(mypet);
-        Pet createdPet = createPetR.as(Pet.class);
-        Assert.assertEquals(createdPet.getId(), mypet.getId());
+    @Step("Then I assert the pet that has been retrieved with the get method is the same pet that has been created")
```

Comment:
Please use shorter descriptions. Is not relevant to know the pet is retrieved using a get method. In this method you verify that the name of a pet with a given id.

Reply:
If you want to check exactly what is in the step name, consider that the pet may have the name you specified but not the rest of the details.
A correct way to check that is to send the pet as parameter being a Pet object and assert the two Pet objects( the one retrieved using the get method and the one sent as parameter